packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -206,6 +206,10 @@ export type TSelectableSearchInputProps = {
    * Show submit button in the input
    */
   showSubmitButton?: boolean;
+  /**
+   * array of selector options that can be used to pass `data-*` props selectors to the options dropdown
+   */
+  optionSelector?: TOptionSelector[];
```

Comment:
I think the name of this prop might be misleading as you should be able to pass any `data-*` attribute and also would be difficult, from the consumer point of view, to understand how those attributes can be forwarded to the select or to the input.

My proposal would be to create two new props for this purpose:
* inputDataProps
* selectDataProps

We can also keep the raw `data-*` props we're currently receiving and forwarding to the `input` element for backwards compatibility.

This way, consumers who want to provide `data` attributes would need to explicitly indicate where they want them to be forwarded.

Example:
```ts
<SelectableSearchInput
  ...
  selectDataProps={{
    'tracking': 'products-search-type'
  }}
  inputDataProps={{
    'tracking': 'product-search-value' 
  }}
  ...
>
```

Then we would transform those values to HTML `data` attributes in the corresponding elements.

@emmenko would that work for you?

Reply:
This makes so much sense. I will refactor once we conclude on this approach

Thank you 🙏🏾

Reply:
Makes sense to me, thanks for the suggestion! 🤗

Reply:
updated here: https://github.com/commercetools/ui-kit/pull/2678/commits/79399cc59a7f211ce9b969875c4b515f8b6c24fe

Thank you